lib/components/desktop/wallpaper_picker.dart

```diff
@@ -97,58 +99,80 @@ class _WallpaperPickerState extends State<WallpaperPicker>
+                                child: Stack(
+                                  children: [
+                                    Positioned.fill(
+                                      child: ResourceImage(
+                                        resource: ImageResource(
+                                          type: ImageResourceType.network,
+                                          value: snapshot
+                                              .data![index]!.downloadUrl,
+                                        ),
+                                        fit: BoxFit.cover,
+                                      ),
+                                    ),
+                                    if (service.wallpaper ==
+                                        ImageResource(
+                                          type: ImageResourceType.network,
+                                          value: snapshot
+                                              .data![index]!.downloadUrl,
+                                        ))
+                                      Positioned(
+                                        bottom: 5,
+                                        right: 5,
+                                        child: CircleAvatar(
+                                          backgroundColor: Theme.of(context)
+                                              .colorScheme
+                                              .secondary,
+                                          foregroundColor: Colors.white,
+                                          child: const Icon(Icons.check),
+                                        ),
+                                      )
+                                    else
+                                      const SizedBox.shrink(),
+                                  ],
                                 ),
                               ),
-                              if (service.wallpaper == builtinWalls[index])
-                                Positioned(
-                                  bottom: 5,
-                                  right: 5,
-                                  child: CircleAvatar(
-                                    backgroundColor:
-                                        Theme.of(context).colorScheme.secondary,
-                                    foregroundColor: Colors.white,
-                                    child: const Icon(Icons.check),
-                                  ),
-                                )
-                              else
-                                const SizedBox.shrink(),
-                            ],
-                          ),
-                        ),
+                            );
+                          },
+                        );
+                      } else if (snapshot.hasError) {
+                        return const Text(
+                          "Error: Failed to fetch data from dahliaOS' GitHub Wallpaper repository API.",
```

Comment:
does this handle github rate-limiting error?

Reply:
Unauthenticated, GitHub allows for 600 requests per originating IP address per hour. While I find that hard to be exceeded (that's 10 requests per 1 minute), I was just thinking of introducing another service like the recentWallpapers one that would store the images if the API errored out for any reason, or better yet, it would not even call the API if the cache is up to date.